cluster/ci/origin/logging-infra/finished.go

```diff
@@ -0,0 +1,64 @@
+package main
+
+import (
+	"time"
+
+	"os"
+	"path/filepath"
+
+	"cloud.google.com/go/storage"
+)
+
+type Finished struct {
+	// Timestamp is the time at which we finish
+	// execution, in seconds past the UNIX epoch
+	Timestamp int64 `json:"timestamp"`
+
+	// Passed is the result of the build
+	Passed bool `json:"passed"`
+}
+
+func UploadFinishedData(processLog, metadataFile, artifactDir string, passed bool, gcsBucket *storage.BucketHandle) error {
+	metadata, err := loadMetadata(metadataFile)
+	if err != nil {
+		return err
+	}
+
+	finishedFile, err := writeFinishedMetadata(passed)
+	if err != nil {
+		return err
+	}
+
+	gcsPath := metadata.GcsPath()
+	uploadTargets := map[string]string{
+		finishedFile: gcsPath + "/finished.json",
+		// TODO(skuznets): we want to stream this log during the run
```

Comment:
No, artifacts+failures are definitely a job for Gubernator. Log streaming otoh seems like something that makes more sense for prow to handle. How can we stream logs in Gubernator when all artifacts are pushed at the end of the job/stage of a job?

ps. there is already an issue for adding log streaming in prow: https://github.com/kubernetes/test-infra/issues/3633